core/src/test/java/org/neo4j/ogm/index/LookupByPrimaryIndexTests.java

```diff
@@ -0,0 +1,87 @@
+package org.neo4j.ogm.index;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import org.neo4j.ogm.domain.cineasts.annotated.User;
+import org.neo4j.ogm.domain.cineasts.partial.Actor;
+import org.neo4j.ogm.index.domain.Invoice;
+import org.neo4j.ogm.session.Neo4jException;
+import org.neo4j.ogm.session.Session;
+import org.neo4j.ogm.session.SessionFactory;
+import org.neo4j.ogm.testutil.MultiDriverTestClass;
+
+/**
+ * @author Mark Angrish
+ */
+public class LookupByPrimaryIndexTests extends MultiDriverTestClass {
+
+    @Test
+    public void loadUsesPrimaryIndexWhenPresent() {
+
+        SessionFactory sessionFactory = new SessionFactory("org.neo4j.ogm.domain.cineasts.annotated");
+        final Session session = sessionFactory.openSession();
+
+        User user1 = new User("login1", "Name 1", "password");
+        session.save(user1);
+
+        final Session session2 = sessionFactory.openSession();
+
+        final User retrievedUser1 = session2.load(User.class, "login1");
+        assertNotNull(retrievedUser1);
+        assertEquals(user1.getLogin(), retrievedUser1.getLogin());
+    }
+
+    @Test
+    public void loadUsesGraphIdWhenPrimaryIndexNotPresent() {
+
+        SessionFactory sessionFactory = new SessionFactory("org.neo4j.ogm.domain.cineasts.partial");
+        final Session session = sessionFactory.openSession();
+
+        Actor actor = new Actor("David Hasslehoff");
+        session.save(actor);
+
+        final Long id = actor.getId();
+
+        final Session session2 = sessionFactory.openSession();
+
+        final Actor retrievedActor = session2.load(Actor.class, id);
+        assertNotNull(retrievedActor);
+        assertEquals(actor.getName(), retrievedActor.getName());
+
+    }
+
+    @Test(expected = Neo4jException.class)
+    public void exceptionRaisedWhenLookupIsDoneWithGraphIdAndThereIsAPrimaryIndexPresent() {
```

Comment:
> The spec says that if there is a primary index present it should be used as the id when doing a session.load()

So in this case, because the type of the @graphId field (Long) is different from the type of the @index field (UUID), we can detect that it must be an incorrect parameter? In the general case, if the type of the @index is say `String` and the user chooses another String field to use for lookup, we won't be able to detect this type problem?

If that's right, then my confusion arose because of the test method name, which specifically mentions graphId, but which presumably could be any non-UUID value.